upup/pkg/fi/cloudup/awstasks/iaminstanceprofilerole.go

```diff
@@ -117,6 +119,13 @@ type terraformIAMInstanceProfile struct {
 }
 
 func (_ *IAMInstanceProfileRole) RenderTerraform(t *terraform.TerraformTarget, a, e, changes *IAMInstanceProfileRole) error {
+	shared := fi.BoolValue(e.Shared)
+	if shared {
+		// Not terraform owned / managed
+		// We won't apply changes, but our validation (kops update) will still warn
```

Comment:
Ah - this type is good.  We do need Shared for terraform / cloudformation, but we do want validation.